### PR TITLE
[Plugin] Add more stats to Peep interface

### PR DESCRIPTION
**Pull request by IntelOrca**
* Add id to the Entity interface.
* Add a few more stats to the Peep entity interface.

### CODE REVIEW

File: distribution/openrct2.d.ts [hunk 1]
```diff
@@ -599,14 +603,90 @@ declare global {
...
+
+        /**
+         * The target nausea value. Nausea will increase / decrease slowly towards this value.
+         */
+        nauseaTarget : number;
+
+        /**
+         * How hungry the guest is between 0 and 255. Lower is more hungry.
+         */
+        hunger: number;
```
Comment (Gymnasiast) [inline distribution/openrct2.d.ts:654]: A minor nitpick, but you seem to use inconsistent formatting (you use both `hunger: number;` and ``hunger : number;`).

File: src/openrct2/scripting/ScEntity.hpp [hunk 2]
```diff
@@ -158,6 +183,220 @@ namespace OpenRCT2::Scripting
...
+        }
+        void maxIntensity_set(uint8_t value)
+        {
+            ThrowIfGameStateNotMutable();
+            auto peep = GetPeep();
+            if (peep != nullptr)
+            {
+                auto minIntensity = peep->intensity & 0x0F;
+                auto maxIntensity = std::min<uint8_t>(value, 15);
+                peep->intensity = (maxIntensity << 4) | minIntensity;
```
Comment (Gymnasiast) [inline src/openrct2/scripting/ScEntity.hpp:366]: I prefer if these implementation details were abstracted away by implementing them as methods on the `Guest` struct. The scripting layer should not have in-depth knowledge of implementation details of the Guest struct, IMO.

It is okay if you address this is another PR, but in that case: please create an issue as well.

**Comment by duncanspumpkin** [comment]
What are we going to do for stats that share fields with staff stats?

**Comment by IntelOrca** [comment]
@duncanspumpkin Not implemented any of those yet, but getEntity could always return a sub object: Guest or Staff which works fine. There might be a problem if you can change between guest and staff in the script.

File: src/openrct2/peep/Peep.h [hunk 3]
```diff
@@ -541,6 +542,58 @@ struct rct_peep_thought
...
+    uint8_t _value{};
+
+public:
+    explicit IntensityRange(uint8_t value)
+        : _value(value)
+    {
+    }
+
+    IntensityRange(uint8_t min, uint8_t max)
+        : _value(std::min<uint8_t>(min, 15) | (std::max<uint8_t>(min, 15) << 4))
```
Comment (Gymnasiast) [inline src/openrct2/peep/Peep.h:557]:
```suggestion
        : _value(std::min<uint8_t>(min, 15) | (std::min<uint8_t>(max, 15) << 4))
```

File: src/openrct2-ui/windows/Guest.cpp [hunk 4]
```diff
@@ -1528,12 +1528,12 @@ void window_guest_stats_paint(rct_window* w, rct_drawpixelinfo* dpi)
     y += LIST_ROW_HEIGHT;
 
     // Intensity
-    int32_t intensity = peep->intensity / 16;
+    auto intensity = peep->intensity.GetMaximum();
```
Comment (Gymnasiast) [inline src/openrct2-ui/windows/Guest.cpp:1531]: Could you rename this variable to `maxIntensity`?

**Pull request merged**